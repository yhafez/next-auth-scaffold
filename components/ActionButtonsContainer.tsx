// Path: ./components/ActionButtonsContainer.tsx
import { ReactNode } from 'react'
import { Box } from '@mui/material'

export interface ActionButtonsContainerProps {
	name: string
	children: ReactNode
}

export default function ActionButtonsContainer({ name, children }: ActionButtonsContainerProps) {
	return (
		<Box
			id={`${name}-buttons-container`}
			sx={{
				display: 'flex',
				flexDirection: 'row',
				justifyContent: 'space-around',
				alignItems: 'center',
				gap: 2,
				margin: '8px 4px',
				width: '80%',
			}}
		>
			{children}
		</Box>
	)
}
