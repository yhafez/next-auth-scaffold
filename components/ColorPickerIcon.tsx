// Path: ./components/ColorPicker.tsx
import { useState, MouseEvent, useCallback, useEffect } from 'react'
import { Box, IconButton, Typography } from '@mui/material'
import { ColorLens } from '@mui/icons-material'
import { darken, lighten } from 'color2k'

import { ColorPickerPopover } from '.'
import { useBoundStore } from '../store'
import {
	getContrastColor,
	getSecondaryColor,
	mixColors,
	findOptimalOverlayOpacity,
	isOverlayNecessary,
} from '../utils/helpers'

export interface ColorPickerProps {
	name: string
	selectedInit?: boolean
}

export default function ColorPickerIcon({ name, selectedInit }: ColorPickerProps) {
	const { darkMode, customPalette, setCustomPalette } = useBoundStore()

	const [anchorEl, setAnchorEl] = useState<HTMLButtonElement | null>(null)
	const [selected, setSelected] = useState(selectedInit ?? false)

	const handleColorChange = useCallback((color: string) => {
		const contrast = getContrastColor(color)
		let mixedColorPrimaryMain = color
		if (isOverlayNecessary(color, contrast, 7)) {
			const overlayColor = contrast === 'black' ? '#FFF' : '#000'
			const overlayOpacity = findOptimalOverlayOpacity(contrast, overlayColor, color, 7)
			mixedColorPrimaryMain = mixColors(color, overlayColor, overlayOpacity)
		}
		let mixedColorPrimaryLight = lighten(mixedColorPrimaryMain, 0.2)
		if (isOverlayNecessary(mixedColorPrimaryLight, contrast, 7)) {
			const overlayColor = contrast === 'black' ? '#FFF' : '#000'
			const overlayOpacity = findOptimalOverlayOpacity(
				contrast,
				overlayColor,
				mixedColorPrimaryLight,
				7,
			)
			mixedColorPrimaryLight = mixColors(mixedColorPrimaryLight, overlayColor, overlayOpacity)
		}
		let mixedColorPrimaryDark = darken(mixedColorPrimaryMain, 0.2)
		if (isOverlayNecessary(mixedColorPrimaryDark, contrast, 7)) {
			const overlayColor = contrast === 'black' ? '#FFF' : '#000'
			const overlayOpacity = findOptimalOverlayOpacity(
				contrast,
				overlayColor,
				mixedColorPrimaryDark,
				7,
			)
			mixedColorPrimaryDark = mixColors(mixedColorPrimaryDark, overlayColor, overlayOpacity)
		}

		const secondaryColor = getSecondaryColor(mixedColorPrimaryMain)
		const secondaryColorContrast = getContrastColor(secondaryColor)
		let mixedColorSecondaryMain = secondaryColor
		if (isOverlayNecessary(secondaryColor, secondaryColorContrast, 7)) {
			const overlayColor = secondaryColorContrast === 'black' ? '#FFF' : '#000'
			const overlayOpacity = findOptimalOverlayOpacity(
				secondaryColorContrast,
				overlayColor,
				secondaryColor,
				7,
			)
			mixedColorSecondaryMain = mixColors(secondaryColor, overlayColor, overlayOpacity)
		}
		let mixedColorSecondaryLight = lighten(mixedColorSecondaryMain, 0.2)
		if (isOverlayNecessary(mixedColorSecondaryLight, secondaryColorContrast, 7)) {
			const overlayColor = secondaryColorContrast === 'black' ? '#FFF' : '#000'
			const overlayOpacity = findOptimalOverlayOpacity(
				secondaryColorContrast,
				overlayColor,
				mixedColorSecondaryLight,
				7,
			)
			mixedColorSecondaryLight = mixColors(mixedColorSecondaryLight, overlayColor, overlayOpacity)
		}
		let mixedColorSecondaryDark = darken(mixedColorSecondaryMain, 0.2)
		if (isOverlayNecessary(mixedColorSecondaryDark, secondaryColorContrast, 7)) {
			const overlayColor = secondaryColorContrast === 'black' ? '#FFF' : '#000'
			const overlayOpacity = findOptimalOverlayOpacity(
				secondaryColorContrast,
				overlayColor,
				mixedColorSecondaryDark,
				7,
			)
			mixedColorSecondaryDark = mixColors(mixedColorSecondaryDark, overlayColor, overlayOpacity)
		}

		const newPalette = {
			...customPalette,
			primary: {
				main: mixedColorPrimaryMain,
				contrastText: contrast,
				light: mixedColorPrimaryLight,
				dark: mixedColorPrimaryDark,
			},
			secondary: {
				main: mixedColorSecondaryMain,
				contrastText: secondaryColorContrast,
				light: mixedColorSecondaryLight,
				dark: mixedColorSecondaryDark,
			},
		}

		setCustomPalette(newPalette)
	}, [])

	useEffect(() => {
		const customPalette = localStorage.getItem('customPalette')
		if (customPalette) handleColorChange(customPalette)
	}, [handleColorChange])

	useEffect(() => {
		setSelected(selectedInit ?? false)
	}, [selectedInit])

	const open = Boolean(anchorEl)
	const id = open ? 'simple-popover' : undefined

	const handleClick = (event: MouseEvent<HTMLButtonElement>) => {
		setAnchorEl(event.currentTarget)
	}

	const handleClose = () => {
		setSelected(false)
		setAnchorEl(null)
	}

	return (
		<>
			<Box
				id={`${name}-color-picker-container`}
				sx={{
					display: 'flex',
					flexDirection: 'column',
					alignItems: 'center',
					justifyContent: 'center',
					p: 0,
					height: '100%',
					width: '100%',
					cursor: 'pointer',
				}}
			>
				<IconButton
					id={`${name}-color-picker-button`}
					aria-describedby={id}
					aria-labelledby={`${name}-color-picker-label`}
					onClick={handleClick}
				>
					<ColorLens
						id={`${name}-color-picker-icon`}
						sx={{
							color: selected
								? darkMode
									? 'primary.light'
									: 'primary.dark'
								: 'primary.contrastText',
							fontSize: '1.5rem',
						}}
						aria-labelledby={`${name}-color-picker-label`}
						role="img"
					/>
				</IconButton>
				<label htmlFor={`${name}-color-picker-button`}>
					<Typography
						id={`${name}-color-picker-label`}
						variant="body2"
						sx={{
							textAlign: 'center',
							fontWeight: 500,
							cursor: 'pointer',
							color: selected
								? darkMode
									? 'primary.light'
									: 'primary.dark'
								: 'primary.contrastText',
						}}
					>
						Theme Color
					</Typography>
				</label>
			</Box>
			<ColorPickerPopover
				name={name}
				anchorEl={anchorEl}
				handleClose={handleClose}
				anchorOrigin={{
					vertical: 'bottom',
					horizontal: 'center',
				}}
				transformOrigin={{
					vertical: 'top',
					horizontal: 'center',
				}}
				handleColorChange={handleColorChange}
			/>
		</>
	)
}
