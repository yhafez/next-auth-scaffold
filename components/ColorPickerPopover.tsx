// Path: ./components/ColorPickerPopover.tsx

import { useState, useCallback } from 'react'
import { Popover, PopoverOrigin } from '@mui/material'
import { useBoundStore } from '../store'
import { ChromePicker } from 'react-color'

export interface ColorPickerPopoverProps {
	name: string
	open: boolean
	anchorEl: HTMLButtonElement | null
	handleClose: () => void
	handleColorChange: (color: string) => void
	anchorOrigin?: PopoverOrigin
	transformOrigin?: PopoverOrigin
}

const ColorPickerPopover = ({
	name,
	open,
	anchorEl,
	handleClose,
	handleColorChange,
	anchorOrigin,
	transformOrigin,
}: ColorPickerPopoverProps) => {
	const { customPalette } = useBoundStore()

	const [color, setColor] = useState(customPalette.primary.main)

	const handleChange = useCallback(
		(color: any) => {
			setColor(color.hex)
			handleColorChange(color.hex)
		},
		[handleColorChange],
	)

	return (
		<Popover
			id={`${name}-color-picker-popover`}
			open={open}
			anchorEl={anchorEl}
			onClose={handleClose}
			anchorOrigin={anchorOrigin}
			transformOrigin={transformOrigin}
			sx={{
				cursor: 'pointer',
				'& .MuiPopover-paper': {
					padding: 2,
					backgroundColor: 'background.paper',
					color: 'text.primary',
				},
			}}
		>
			<ChromePicker color={color} onChange={color => handleChange(color)} disableAlpha />
		</Popover>
	)
}

export default ColorPickerPopover
