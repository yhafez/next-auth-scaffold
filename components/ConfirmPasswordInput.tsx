// Path: ./components/ConfirmPasswordInput.tsx
import { Dispatch, SetStateAction, KeyboardEvent } from 'react'

import PasswordInput from './PasswordInput'

export interface ConfirmPasswordInputProps {
	name: string
	password: string
	setPassword: Dispatch<SetStateAction<string>>
	confirmPassword: string
	setConfirmPassword: Dispatch<SetStateAction<string>>
	disabled?: boolean
	handleEnter?: (e: KeyboardEvent) => void
}

export default function ConfirmPasswordInput({
	name,
	password,
	setPassword,
	confirmPassword,
	setConfirmPassword,
	disabled = false,
	handleEnter,
}: ConfirmPasswordInputProps) {
	return (
		<>
			<PasswordInput
				name={name}
				value={password}
				setValue={setPassword}
				disabled={disabled}
				handleEnter={handleEnter}
			/>
			<PasswordInput
				name={`${name}`}
				value={confirmPassword}
				setValue={setConfirmPassword}
				disabled={disabled}
				isConfirmPassword={true}
				handleEnter={handleEnter}
			/>
		</>
	)
}
