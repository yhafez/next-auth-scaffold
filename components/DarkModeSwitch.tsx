// Path: ./components/DarkModeSwitch.tsx
import { useEffect, useState } from 'react'
import { Box, Switch, Typography } from '@mui/material'
import { WbSunny, Brightness2 } from '@mui/icons-material'

import { useBoundStore } from '../store'

export interface DarkModeSwitchProps {
	name: string
}

export default function DarkModeSwitch({ name }: DarkModeSwitchProps) {
	const { darkMode, setDarkMode } = useBoundStore()
	const [selected, setSelected] = useState(false)

	const handleDarkMode = () => {
		localStorage.setItem('darkMode', JSON.stringify(!darkMode))
		setDarkMode(!darkMode)
	}

	useEffect(() => {
		const darkMode = localStorage.getItem('darkMode') === 'true'
		setDarkMode(darkMode)
	}, [])

	const handleEnter = (e: React.KeyboardEvent<HTMLButtonElement>) => {
		if (e.key === 'Enter') {
			handleDarkMode()
		}
	}

	return (
		<Box
			id={`${name}-dark-mode-switch-container`}
			onMouseEnter={() => setSelected(true)}
			onMouseLeave={() => setSelected(false)}
			onFocus={() => setSelected(true)}
			onBlur={() => setSelected(false)}
			sx={{
				textAlign: 'center',
				display: 'flex',
				flexDirection: 'column',
				justifyContent: 'center',
				alignItems: 'center',
				height: '100%',
				width: '100%',
			}}
		>
			<Switch
				id={`${name}-dark-mode-switch`}
				checked={darkMode}
				onChange={handleDarkMode}
				onKeyDown={handleEnter}
				inputProps={{
					'aria-checked': darkMode,
					'aria-labelledby': `${name}-dark-mode-switch-label-${darkMode ? 'dark' : 'light'}`,
					role: 'switch',
				}}
				sx={{
					fontSize: '1.5rem',
					color: selected ? (darkMode ? 'primary.light' : 'primary.dark') : 'primary.contrastText',
					'& .MuiSwitch-switchBase': {
						color: 'primary.contrastText',
						'&.Mui-checked': {
							color: 'primary.contrastText',
						},
						'&.Mui-checked + .MuiSwitch-track': {
							backgroundColor: 'secondary.main',
						},
					},
				}}
				checkedIcon={
					<Brightness2
						id={`${name}-dark-mode-switch-icon-checked`}
						aria-hidden={darkMode ? 'false' : 'true'}
						aria-labelledby={`${name}-dark-mode-switch-label-dark`}
						role="img"
						sx={{
							fontSize: '1.5rem',
							color: selected
								? darkMode
									? 'primary.light'
									: 'primary.dark'
								: 'primary.contrastText',
						}}
					/>
				}
				icon={
					<WbSunny
						id={`${name}-dark-mode-switch-icon-unchecked`}
						aria-hidden={darkMode ? 'true' : 'false'}
						aria-labelledby={`${name}-dark-mode-switch-label-light`}
						role="img"
						sx={{
							fontSize: '1.5rem',
							color: selected
								? darkMode
									? 'primary.light'
									: 'primary.dark'
								: 'primary.contrastText',
						}}
					/>
				}
			/>
			<label htmlFor={`${name}-dark-mode-switch`}>
				<Typography
					id={`${name}-dark-mode-switch-label-${darkMode ? 'dark' : 'light'}`}
					variant="body2"
					sx={{
						fontWeight: 500,
						color: selected
							? darkMode
								? 'primary.light'
								: 'primary.dark'
							: 'primary.contrastText',
						cursor: 'pointer',
					}}
				>
					{darkMode ? 'Dark' : 'Light'}
					<br /> Mode
				</Typography>
			</label>
		</Box>
	)
}
