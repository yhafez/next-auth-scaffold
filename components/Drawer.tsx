// Path: ./components/Drawer.tsx
import { Drawer, IconButton, Box, Typography, useMediaQuery } from '@mui/material'
import { ChevronLeft } from '@mui/icons-material'

import { useBoundStore } from '../store'

export interface DrawerProps {
	open: boolean
	handleDrawerClose: () => void
	children: React.ReactNode
}

export default function DrawerComponent({ open, handleDrawerClose, children }: DrawerProps) {
	const { darkMode, theme } = useBoundStore()
	const isMobile = useMediaQuery(`(max-width:${theme.breakpoints.values.sm}px)`)

	return (
		<Drawer
			id="drawer"
			sx={{
				width: '18em',
				flexShrink: 0,
				overflowY: 'scroll',
				'& .MuiDrawer-paper': {
					width: '18em',
					boxSizing: 'border-box',
					backgroundColor: darkMode ? 'primary.dark' : 'primary.light',
					color: 'primary.contrastText',
					display: 'flex',
					flexDirection: 'column',
					justifyContent: 'flex-start',
				},
			}}
			variant={isMobile ? 'temporary' : 'permanent'}
			open={open}
			anchor="left"
			onClose={handleDrawerClose}
		>
			<>
				{isMobile && (
					<Box
						id="drawer-close-button-container"
						onClick={handleDrawerClose}
						sx={{
							display: 'flex',
							alignItems: 'center',
							justifyContent: 'space-between',
							padding: 1,
							...theme.mixins.toolbar,
							borderBottom: '1px solid rgba(255, 255, 255, 0.12)',
							pt: 2,
							pb: 2,
							px: 0,
							width: '100%',
							height: '2em',
							cursor: 'pointer',
							'&:hover': {
								backgroundColor: 'primary.main',
							},
							'&:focus': {
								backgroundColor: 'primary.main',
							},
							'&:active': {
								backgroundColor: 'primary.main',
							},
						}}
					>
						<IconButton
							id="drawer-close-button"
							sx={{
								color: 'primary.contrastText',
								height: '100%',
								borderRadius: 0,
								'&:hover': {
									backgroundColor: 'transparent',
								},
							}}
						>
							<ChevronLeft
								id="drawer-close-button-icon"
								sx={{ fontSize: 40 }}
								role="img"
								aria-labelledby="drawer-close-button-label-text"
							/>
						</IconButton>
						<label id="drawer-close-button-label" htmlFor="drawer-close-button">
							<Typography
								id="drawer-close-button-label-text"
								sx={{
									fontWeight: 500,
									pr: 2,
									color: 'primary.contrastText',
									ml: 1,
									cursor: 'pointer',
									'&:hover': {
										color: darkMode ? 'primary.light' : 'primary.dark',
										transform: 'scale(1.1)',
									},
									'&:focus': {
										color: darkMode ? 'primary.light' : 'primary.dark',
										transform: 'scale(1.1)',
									},
									'&:active': {
										color: darkMode ? 'primary.light' : 'primary.dark',
										transform: 'scale(1.1)',
									},
								}}
							>
								Close
							</Typography>
						</label>
					</Box>
				)}
				{children}
			</>
		</Drawer>
	)
}
