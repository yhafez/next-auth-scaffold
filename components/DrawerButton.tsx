// Path: ./components/DrawerButton.tsx
import { ReactNode, MouseEvent } from 'react'
import Link from 'next/link'
import { useRouter } from 'next/router'
import { Link as MuiLink, IconButton, Typography } from '@mui/material'

export interface DrawerButtonProps {
	name: string
	label: string
	icon: ReactNode
	handleClick?: (e: MouseEvent<HTMLButtonElement>) => void
	reverse?: boolean
	href?: string
}

export default function DrawerButton({
	name,
	label,
	icon,
	handleClick,
	reverse,
	href,
}: DrawerButtonProps) {
	const router = useRouter()
	const isSelected = router.pathname === href ? true : false

	if (href) {
		return (
			<MuiLink
				id={`${name}-drawer-link`}
				component={Link}
				href={href}
				aria-current={isSelected ? 'page' : undefined}
				title={label}
				aria-labelledby={`${name}-drawer-button-label`}
				sx={{
					textDecoration: 'none',
					color: 'primary.contrastText',
					padding: 0,
					display: 'flex',
					alignItems: 'center',
					justifyContent: 'space-between',
					width: '100%',
					borderRadius: 0,
					backgroundColor: isSelected ? 'primary.main' : 'transparent',
					p: 4,
					'&:hover': {
						backgroundColor: 'primary.main',
					},
					'&:focus': {
						backgroundColor: 'primary.main',
					},
					'&:active': {
						backgroundColor: 'primary.main',
					},
				}}
			>
				{reverse && icon}
				<Typography
					id={`${name}-drawer-button-label`}
					variant="body1"
					sx={{
						fontSize: '1.2rem',
						fontWeight: 'bold',
					}}
				>
					{label}
				</Typography>
				{!reverse && icon}
			</MuiLink>
		)
	}

	return (
		<IconButton
			id={`${name}-drawer-button`}
			onClick={handleClick}
			disableRipple
			sx={{
				color: 'primary.contrastText',
				padding: 0,
				display: 'flex',
				alignItems: 'center',
				justifyContent: 'space-between',
				width: '100%',
				borderRadius: 0,
				backgroundColor: isSelected ? 'primary.main' : 'transparent',
				p: 4,
				'&:hover': {
					backgroundColor: 'primary.main',
				},
				'&:focus': {
					backgroundColor: 'primary.main',
				},
				'&:active': {
					backgroundColor: 'primary.main',
				},
			}}
		>
			{reverse && icon}
			<Typography
				id={`${name}-drawer-button-label`}
				variant="body1"
				sx={{
					fontSize: '1.2rem',
					fontWeight: 'bold',
				}}
			>
				{label}
			</Typography>
			{!reverse && icon}
		</IconButton>
	)
}
