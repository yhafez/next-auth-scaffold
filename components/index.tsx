// Path: ./components/index.tsx
import Layout from './Layout'

import DashboardDrawer from './Drawers/DashboardDrawer'
import SettingsDrawer from './Drawers/SettingsDrawer'

import ColorPickerPopover from './Popovers/ColorPickerPopover'
import DrawerSignOutPopover from './Popovers/DrawerSignOutPopover'

import ActionButtonsContainer from './ActionButtonsContainer'
import ColorPickerIcon from './ColorPickerIcon'
import ConfirmPasswordInput from './ConfirmPasswordInput'
import DarkModeSwitch from './DarkModeSwitch'
import Drawer from './Drawer'
import DrawerButton from './DrawerButton'
import EditableTextField from './EditableTextField'
import EmailInput from './EmailInput'
import Modal from './Modal'
import ModalButton from './ModalButton'
import ModalNote from './ModalNote'
import NavigationButton from './NavigationButton'
import PasswordInput from './PasswordInput'
import SubmitButton from './SubmitButton'
import VisibilityIconButton from './VisibilityIconButton'

export {
	Layout,
	DashboardDrawer,
	SettingsDrawer,
	ColorPickerPopover,
	DrawerSignOutPopover,
	ActionButtonsContainer,
	ColorPickerIcon,
	ConfirmPasswordInput,
	DarkModeSwitch,
	Drawer,
	DrawerButton,
	EditableTextField,
	EmailInput,
	Modal,
	ModalButton,
	ModalNote,
	NavigationButton,
	PasswordInput,
	SubmitButton,
	VisibilityIconButton,
}
