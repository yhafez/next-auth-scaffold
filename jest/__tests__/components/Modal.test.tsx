// Path: ./jest/__tests__/components/Modal.test.tsx
import { render, findByText, act, screen, waitFor } from '@testing-library/react'
import { axe } from 'jest-axe'

import { Modal } from '../../../components'

describe('Modal', () => {
	it('Should have no accessibility violations', async () => {
		act(() =>
			render(
				<main>
					<Modal name="test test" loading={false} error="" small={false}>
						<div>Children</div>
					</Modal>
				</main>,
			),
		)

		waitFor(async () => expect(await axe(screen.getByRole('main'))).toHaveNoViolations())
	})

	it('Should have a Box element with id "test-test-modal-container"', async () => {
		const { container } = render(
			<Modal name="test test" loading={false} error="" small={false}>
				<div>Children</div>
			</Modal>,
		)

		const test = container.querySelector('#test-test-modal-container')
		expect(test).toBeTruthy()
	})

	it('Should have a Typography element with id "test-test-modal-title-word-0"', async () => {
		const { container } = render(
			<Modal name="test test" loading={false} error="" small={false}>
				<div>Children</div>
			</Modal>,
		)

		const test = container.querySelector('#test-test-modal-title-word-0')
		expect(test).toBeTruthy()
	})

	it('Should have a Typography element with text "test " and a Typography element with text "test', async () => {
		const { container } = render(
			<Modal name="test test" loading={false} error="" small={false}>
				<div>Children</div>
			</Modal>,
		)

		const test = container.querySelector('#test-test-modal-title-word-0')?.textContent
		expect(test).toBe('Test ')
		const test2 = container.querySelector('#test-test-modal-title-word-1')?.textContent
		expect(test2).toBe('Test')
	})

	it('Should have a Box element with id "test-test-modal-notification-container"', async () => {
		const { container } = render(
			<Modal name="test test" loading={false} error="" small={false}>
				<div>Children</div>
			</Modal>,
		)

		const test = container.querySelector('#test-test-modal-notification-container')
		expect(test).toBeTruthy()
	})

	it('Should have a Box element with id "test-test-modal-form"', async () => {
		const { container } = render(
			<Modal name="test test" loading={false} error="" small={false}>
				<div>Children</div>
			</Modal>,
		)

		const test = container.querySelector('#test-test-modal-form')
		expect(test).toBeTruthy()
	})

	it('Should have a Div element with text "Children"', async () => {
		const { container } = render(
			<Modal name="test test" loading={false} error="" small={false}>
				<div>Children</div>
			</Modal>,
		)

		const test = await findByText(container, 'Children')
		expect(test).toBeTruthy()
	})

	it('Should have a CircularProgress element with id "test-test-modal-loading"', async () => {
		const { container } = render(
			<Modal name="test test" loading={true} error="" small={false}>
				<div>Children</div>
			</Modal>,
		)

		const test = container.querySelector('#test-test-modal-loading')
		expect(test).toBeTruthy()
	})

	it('Should have a Typography element with id "test-test-modal-error"', async () => {
		const { container } = render(
			<Modal name="test test" loading={false} error="test error" small={false}>
				<div>Children</div>
			</Modal>,
		)

		const test = container.querySelector('#test-test-modal-error')
		expect(test).toBeTruthy()
	})

	it('Should have a Typography element with text "test error"', async () => {
		const { container } = render(
			<Modal name="test test" loading={false} error="test error" small={false}>
				<div>Children</div>
			</Modal>,
		)

		const test = await findByText(container, 'test error')
		expect(test).toBeTruthy()
	})

	it('Should not have a Typography element with id "test-test-modal-error"', async () => {
		const { container } = render(
			<Modal name="test test" loading={true} error="test error" small={false}>
				<div>Children</div>
			</Modal>,
		)

		const test = container.querySelector('#test-test-modal-error')
		expect(test).toBeFalsy()
	})

	it('Should have a CircularProgress element with id "test-test-modal-loading"', async () => {
		const { container } = render(
			<Modal name="test test" loading={true} error="test error" small={false}>
				<div>Children</div>
			</Modal>,
		)

		const test = container.querySelector('#test-test-modal-loading')
		expect(test).toBeTruthy()
	})

	it('Should match snapshot', () => {
		const { container } = render(
			<Modal name="test test" loading={false} error="" small={false}>
				<div>Children</div>
			</Modal>,
		)

		expect(container).toMatchSnapshot()
	})
})
