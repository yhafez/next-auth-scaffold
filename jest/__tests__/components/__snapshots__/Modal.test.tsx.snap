// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-1054lpv"
    id="test-test-modal-container"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-2yvvac-MuiTypography-root"
      id="test-test-modal-title-word-0"
    >
      Test 
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-2yvvac-MuiTypography-root"
      id="test-test-modal-title-word-1"
    >
      Test
    </p>
    <div
      class="MuiBox-root css-j4gqgf"
      id="test-test-modal-notification-container"
    />
    <div
      class="MuiBox-root css-1knh30t"
      id="test-test-modal-form"
    >
      <div>
        Children
      </div>
    </div>
  </div>
</div>
`;
